### PR TITLE
Issue #2: Add unit tests

### PR DESCRIPTION
**Pull request by lmorchard**
* Unit test for platform.js

* Unit test and example directory structure for findSimulators()

* Add a simple .travis.yml

* Add VIM swapfiles to .gitignore

* Add an .editorconfig with 2-space indentation rules

### CODE REVIEW

**Comment by lmorchard** [comment]
Sadly, travis-ci won't run the darwin & win32 tests, though I can vouch for them having passed on my laptop and the win8 machine in my living room.

**Comment by tofumatt** [comment]
Tangential to the PR, but what should our solution be for running the tests on multiple platforms? Different post-commit hooks or something? How do multi-platform things handle it, or does it have to be custom?

**Comment by lmorchard** [comment]
That's a good question. I'm not sure I've ever heard of CI in the cloud for node on darwin & win32 platforms. In fact, the only CI farm I can think of that includes darwin & win32 is the one Mozilla uses for Firefox. Not that there isn't one, but it could probably do with some research.

**Comment by lmorchard** [comment]
Oh wait, [travis apparently has OS X](http://docs.travis-ci.com/user/osx-ci-environment/). Time for more reading!

**Comment by lmorchard** [comment]
Looks like there's a "[Testing Your Project on Multiple Operating Systems](http://docs.travis-ci.com/user/multi-os/)" feature in beta on Travis, but it only includes Linux & OS X and needs to be explicitly enabled by a repo owner.

**Comment by lmorchard** [comment]
So, IMO, getting this working automatically on multiple OSes would be a good thing. But, it probably shouldn't be a blocker for getting Travis working for at least linux

**Comment by tofumatt** [comment]
Oh, agreed, and testing on Linux should buy us a fair bit. We can continue this discussion elsewhere if you like; I’ll actually review the code now :-)

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# http://editorconfig.org
```
Comment (tofumatt) [inline .editorconfig]: At some point we might want these in the build tools? I'm not sure how to handle that, but all projects should share these settings.

File: tests/unit/data/darwin/Library/Application Support/Firefox/Profiles/asdflulz.Release/extensions/fxos_1_3_simulator@mozilla.org/b2g/B2G.app/Contents/MacOS/b2g-bin [hunk 2]
```diff
@@ -0,0 +1 @@
+Shh, I am pretending to be a simulator.
```
Comment (tofumatt) [inline tests/unit/data/darwin/Library/Application Support/Firefox/Profiles/asdflulz.Release/extensions/fxos_1_3_simulator@mozilla.org/b2g/B2G.app/Contents/MacOS/b2g-bin]: What are these here for?

File: tests/unit/test.index.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      { version: '1.5',
+        bin: '%PROFILEDIR%/ou812wow.Nightly/extensions/fxos_1_5_simulator@mozilla.org/%BINARY%',
+        profile: '%PROFILEDIR%/ou812wow.Nightly/extensions/fxos_1_5_simulator@mozilla.org/profile' },
+      { version: '2.2',
+        bin: '%PROFILEDIR%/ou812wow.Nightly/extensions/fxos_2_2_simulator@mozilla.org/%BINARY%',
+        profile: '%PROFILEDIR%/ou812wow.Nightly/extensions/fxos_2_2_simulator@mozilla.org/profile' }
+    ];
+
+    // Replace platform-specific placeholders in the expected data
+    for (var idx = 0; idx < expected.length; idx++) {
```
Comment (tofumatt) [inline tests/unit/test.index.js]: Oh god the two/three letter variables. Could they be more verbose? I find a lot of short iterators hard to read. But then I am an old man.

upf1

File: tests/unit/test.index.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        }
+        if (process.platform === 'win32') {
+          // HACK: Correct expected path when running tests on win32
+          item[ki] = item[ki].replace(/\//g,'\\');
+        }
+      }
+    }
+
+    findSimulators().then(function(result) {
+      test.deepEqual(result, expected);
```
Comment (tofumatt) [inline tests/unit/test.index.js]: This compares all of an object's key/values deeply I assume? A comment might be nice :wink:

File: tests/unit/test.platform.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    });
+    test.done();
+  }
+
+};
+
+function testDeepEqualPaths(test, result, expected) {
+  if (process.platform === 'win32') {
+    // HACK: Correct expected path when running tests on win32
+    expected.firefoxProfilesDir = expected.firefoxProfilesDir.replace(/\//g,'\\');
```
Comment (tofumatt) [inline tests/unit/test.platform.js]: Don't the Windows paths already have `\\` in them?

**Comment by tofumatt** [comment]
r+wc, basically. Looks good!

File: tests/unit/data/darwin/Library/Application Support/Firefox/Profiles/asdflulz.Release/extensions/fxos_1_3_simulator@mozilla.org/b2g/B2G.app/Contents/MacOS/b2g-bin (hunk 2 above)
Comment (lmorchard) [inline tests/unit/data/darwin/Library/Application Support/Firefox/Profiles/asdflulz.Release/extensions/fxos_1_3_simulator@mozilla.org/b2g/B2G.app/Contents/MacOS/b2g-bin]: They are the files pretending to be simulators that the code finds as part of the tests.

File: tests/unit/test.platform.js (hunk 5 above)
Comment (lmorchard) [inline tests/unit/test.platform.js]: Yes, but the linux paths don't. Line 40 "tests" the windows code paths under linux, which results in a "/" from path.join. But, when the tests run on win32, that '/' turns into a '\'

Comment (tofumatt) [inline tests/unit/test.platform.js]: Oh, I see, there is the one `/` in the path. That was hard to catch. Alright, carry on!

File: tests/unit/data/darwin/Library/Application Support/Firefox/Profiles/asdflulz.Release/extensions/fxos_1_3_simulator@mozilla.org/b2g/B2G.app/Contents/MacOS/b2g-bin (hunk 2 above)
Comment (tofumatt) [inline tests/unit/data/darwin/Library/Application Support/Firefox/Profiles/asdflulz.Release/extensions/fxos_1_3_simulator@mozilla.org/b2g/B2G.app/Contents/MacOS/b2g-bin]: That checks out. I’m dumb. Continue.

File: tests/unit/test.index.js (hunk 4 above)
Comment (lmorchard) [inline tests/unit/test.index.js]: Yeah, that's just part of the [nodeunit API](https://github.com/caolan/nodeunit#api-documentation), nothing special

File: tests/unit/test.index.js (hunk 3 above)
Comment (lmorchard) [inline tests/unit/test.index.js]: Ugh, yeah I'm lazy about that for quick & dirty bits like this. I can expand the variable names though

Comment (tofumatt) [inline tests/unit/test.index.js]: I would <3 you forever if you did :-)

File: .editorconfig (hunk 1 above)
Comment (lmorchard) [inline .editorconfig]: Hmm, I haven't checked out the build tools repo yet...

Comment (tofumatt) [inline .editorconfig]: It's running your `gulp` tasks for you: https://github.com/mozilla/node-firefox-find-simulators/blob/master/gulpfile.js ^_^

https://github.com/mozilla/node-firefox-build-tools/blob/master/lib/gulp-tasks.js

**Comment by lmorchard** [comment]
Just made a couple of tweaks - expanded variable names in that mess 'o loops I've got up there, and a small comment above the deepEqual

**Pull request merged**